buildrunner/validation/config.py

```diff
@@ -132,6 +138,155 @@ def validate_push(
+                    src_image.append(f"{step.run.image}:latest")
+                else:
+                    src_image.append(step.run.image)
+
+            if step.build and step.build.platforms:
+                is_multi_platform = True
+
+            return src_image, is_multi_platform
+
+        def get_destination_images(step: Step) -> Optional[List[str]]:
+            """
+            Get the destination images from step.push or step.commit
+
+            Args:
+                step (Step): Build step
+
+            Returns:
+                Optional[List[str]]: List of destination images
+            """
+
+            def get_images(
+                pushcommmit: Union[
+                    StepPushCommitDict, str, List[Union[str, StepPushCommitDict]]
+                ],
+            ) -> List[str]:
+                images = []
+                if isinstance(pushcommmit, StepPushCommitDict):
+                    if pushcommmit.tags:
+                        for tag in pushcommmit.tags:
+                            images.append(f"{pushcommmit.repository}:{tag}")
+                    else:
+                        images.append(f"{pushcommmit.repository}:latest")
+                elif isinstance(pushcommmit, str):
+                    if ":" not in pushcommmit:
+                        images.append(f"{pushcommmit}:latest")
+                    else:
+                        images.append(pushcommmit)
+                elif isinstance(pushcommmit, list):
+                    for item in pushcommmit:
+                        if isinstance(item, str):
+                            images.append(item)
+                        elif isinstance(item, StepPushCommitDict):
+                            if item.tags:
+                                for tag in item.tags:
+                                    images.append(f"{item.repository}:{tag}")
+                            else:
+                                images.append(f"{item.repository}:latest")
+                        else:
+                            raise ValueError(
+                                f"Unknown type for step.push: {type(step.push)}"
+                            )
+                return images
+
+            images = []
+            images.extend(get_images(step.push))
+            images.extend(get_images(step.commit))
+            return images
+
+        def validate_multiplatform_are_not_retagged():
+            """
+            Validate multi-platform are not re-tagged
+
+            Args:
+                mp_push_tags (Set[str]): Set of all tags used in multi-platform build steps
+
+            Raises:
+                ValueError | pydantic.ValidationError: If the config file is invalid
+            """
+            step_images = {}
+
+            # Iterate through each step and get information about the images
+            for step_name, step in vals.items():
+                src_image, is_multi_platform = get_source_image(step)
```

Comment:
Why does this need to be calculated/returned when there is a `step.is_multi_platform()` that can be used (see line 302).

Reply:
I should be using `step.is_multi_platform()`, I may have forget that was available. But I do need to keep track of whether this step is multi-platform or not so I can compare the source/dest images in this step with the other steps to see if there are invalid multi-platform `re-tagging`.

On line 302, that is check for specific things within the multi-platform step and is not concerned with other steps.

I fixed this to use `step.is_multi_platform()`, nice catch.

Reply:
Yeah, the check is good, just wanted to get rid of the extra definition of is_multi_platform and returning it from a method that isn't too clear it returns it, etc